docs/basics/part-4-reducing-code-duplication.md

```diff
@@ -1,16 +1,14 @@
-# Reducing code duplication
+To copy the files for [this example ( Part 4 )](https://github.com/earthly/earthly/tree/main/examples/tutorial/go/part4) run
```

Comment:
I feel that this part doesn't really add much. In the previous version we had an excuse like this one, which I think just reinforces that this is a bad example 😄 

> Note that in our case, only the JavaScript version has an example where `FROM +deps` is used in more than one place: both in `build` and in `docker`. Nevertheless, all versions show how dependencies may be separated.

We should either

* Remove this part altogether
* Merge this with part 4 (maybe in a minimal form)
* Come up with a different example that showcases the idea better

I would go with the second option.

Reply:
Moved to the end of part 3